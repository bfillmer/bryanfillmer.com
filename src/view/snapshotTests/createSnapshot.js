/* global test, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import {createStore} from 'redux'
import {Provider} from 'react-redux'

import {reducers} from 'state/reducers'
import {App} from 'view/App'

// Run a snapshot test that renders the entire application with a given Redux store, parsing
// actions exported using redux-dev-tools.
export const createSnapshot = (description, payload) => {
  const actionsArray = JSON.parse(payload)
  // Create a new store to avoid polluting the test.
  const store = createStore(reducers)
  // Replay the actions in order to end up with our needed state for our expected UI visual.
  actionsArray.map(store.dispatch)
  // Perform a full tree render test for this snapshot.
  test(`[SNAP] ${description}`, () => {
    const tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
}
