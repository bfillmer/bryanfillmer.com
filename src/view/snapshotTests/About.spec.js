
import {createSnapshot} from './createSnapshot'

import data from './actionPayloads/About.json'

createSnapshot('About view loads and renders as expected', data.payload)
