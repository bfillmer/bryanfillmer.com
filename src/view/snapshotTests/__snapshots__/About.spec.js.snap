// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAP] About view loads and renders as expected 1`] = `
<div
  className="sc-bxivhb EYjjH"
>
  <header
    className="sc-htpNat evPKAK"
  >
    <img
      alt="logo"
      className="sc-bwzfXH ldyXqg"
      src="logo.svg"
    />
    <h2>
      Welcome to 
      React
    </h2>
  </header>
  <p>
    Head to the 
    <a
      href="/"
      onClick={[Function]}
      onMouseDown={false}
      onTouchStart={false}
      target={undefined}
    >
      Home Page
    </a>
    .
  </p>
</div>
`;
